examples/hotel_ads/add_performance_max_for_travel_goals_campaign.rb

```diff
@@ -0,0 +1,823 @@
+      m.asset_group_asset_operation =
+        client.operation.create_resource.asset_group_asset do |aga|
+          aga.field_type = :HEADLINE
+          aga.asset_group =
+            client.path.asset_group(customer_id, ASSET_GROUP_TEMPORARY_ID)
+          aga.asset = resource_name
+        end
+    end
+  end
+
+  #  Link the description assets.
+  description_asset_resource_names.each do |resource_name|
+    operations << client.operation.mutate do |m|
+      m.asset_group_asset_operation =
+        client.operation.create_resource.asset_group_asset do |aga|
+          aga.field_type = :DESCRIPTION
+          aga.asset_group =
+            client.path.asset_group(customer_id, ASSET_GROUP_TEMPORARY_ID)
+          aga.asset = resource_name
+        end
+    end
+  end
+
+  # [START link_hotel_asset]
+  # Link the previously created hotel property asset to the asset group.
+  # In the real-world scenario, you'd need to do this step several times for
+  # each hotel property asset.
+  operations << client.operation.mutate do |m|
+    m.asset_group_asset_operation =
+      client.operation.create_resource.asset_group_asset do |aga|
+        aga.field_type = :HOTEL_PROPERTY
+        aga.asset_group =
+          client.path.asset_group(customer_id, ASSET_GROUP_TEMPORARY_ID)
+        aga.asset = hotel_property_asset_resource_name
+      end
+  end
+  # [END link_hotel_asset]
+
+  # Creates the rest of required text assets and link them to the asset group.
+  operations +=
+    create_text_assets_for_asset_group(
+      client,
+      customer_id,
+      hotel_asset_suggestion
+    )
+
+  # Creates the image assets and link them to the asset group. Some optional
+  # image assets suggested by the TravelAssetSuggestionService might be created too.
+  operations +=
+    create_image_assets_for_asset_group(
+      client,
+      customer_id,
+      hotel_asset_suggestion
+    )
+
+  operations
+end
+
+# Creates text assets required for an asset group using the suggested hotel text
+# assets. It adds more text assets to fulfill the requirements if the suggested
+# hotel text assets are not enough.
+def create_text_assets_for_asset_group(
+  client,
+  customer_id,
+  hotel_asset_suggestion
+)
+  operations = []
+
+  required_text_asset_counts = deep_copy(MIN_REQUIRED_TEXT_ASSET_COUNTS)
+  required_text_asset_counts.each do |asset_field_type, _count|
+    required_text_asset_counts[asset_field_type] = 0
+  end
+
+  # Creates mutate operations for the suggested text assets except for headlines
+  # and descriptions, which were created previously.
+  if hotel_asset_suggestion.status == :SUCCESS
+    hotel_asset_suggestion.text_assets.each do |hotel_text_asset|
+      if hotel_text_asset.asset_field_type == :HEADLINE ||
+         hotel_text_asset.asset_field_type == :DESCRIPTION
+        # Headlines and descriptions were already created at the first step of
+        # this code example.
+        next
+      end
+
+      puts format(
```

Comment:
Prefer in-line replacement.

`"A text asset with text '#{hotel_text_asset.text}' is ..."`